snuba/migrations/snuba_migrations/functions/0001_functions.py

```diff
@@ -0,0 +1,189 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    AggregateFunction,
+    Array,
+    Column,
+    DateTime,
+    Float,
+    Nested,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+common_columns: List[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("transaction_name", String()),
+    Column("timestamp", DateTime()),
+    Column("depth", UInt(32)),
+    Column("parent_fingerprint", UInt(64)),
+    Column("fingerprint", UInt(64)),
+    Column("symbol", String()),
+    Column("image", String()),
+    Column("filename", String()),
+    Column("is_application", UInt(8)),
+    Column("tags", Nested([Column("key", String()), Column("value", String())])),
+    Column("retention_days", UInt(16)),
+]
+
+raw_columns: List[Column[Modifiers]] = common_columns + [
+    Column("durations", Array(Float(64))),
+    Column("profile_id", UUID()),
+]
+
+agg_columns: List[Column[Modifiers]] = common_columns + [
+    Column("count", AggregateFunction("count", [Float(64)])),
+    Column(
+        "percentiles",
+        AggregateFunction("quantiles(0.5, 0.75, 0.9, 0.95, 0.99)", [Float(64)]),
+    ),
+    Column("min", AggregateFunction("min", [Float(64)])),
+    Column("max", AggregateFunction("max", [Float(64)])),
+    Column("avg", AggregateFunction("avg", [Float(64)])),
+    Column("sum", AggregateFunction("sum", [Float(64)])),
+    Column("worst", AggregateFunction("argMax", [UUID(), Float(64)])),
+    Column("examples", AggregateFunction("groupUniqArray(5)", [UUID()])),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    index_granularity = "2048"
+    storage_set = StorageSetKey.FUNCTIONS
+
+    data_granularity = 60 * 60
+
+    local_raw_table = "functions_raw_local"
+    dist_raw_table = "functions_raw_dist"
+
+    local_materialized_table = "functions_mv_local"
+    dist_materialized_table = "functions_mv_dist"
+
+    local_view_table = "functions_local"
+    dist_view_table = "functions_dist"
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=self.storage_set,
+                table_name=self.local_raw_table,
+                columns=raw_columns,
+                engine=table_engines.MergeTree(
+                    storage_set=self.storage_set,
+                    order_by="(org_id, project_id, transaction_name, timestamp)",
+                    partition_by="(toStartOfInterval(timestamp, INTERVAL 3 day))",
+                    ttl="timestamp + toIntervalDay(7)",
```

Comment:
Do you need 7 days of data? 
How backdated can profile be? Are you going to allow the ingestion of profiles that happened days ago ?